libcxx/test/support/test_macros.h

```diff
@@ -291,17 +291,27 @@ struct is_same<T, T> { enum {value = 1}; };
 // when optimizations are enabled.
 template <class Tp>
 inline Tp const& DoNotOptimize(Tp const& value) {
-    asm volatile("" : : "r,m"(value) : "memory");
-    return value;
+  // The `m` constraint is invalid in the AMDGPU backend.
```

Comment:
I'm not sure what it means but can we handle it?

Reply:
> 'm' A memory operand is allowed, with any kind of address that the machine supports in general. Note that the letter used for the general memory constraint can be re-defined by a back end using the TARGET_MEM_CONSTRAINT macro.

According to the GNU docs, but I just found it didn't work in practice.

Reply:
That's stuff they went and implemented on their own without telling us

Reply:
Figured it'd not really necessary for this case anyway, since anything we care about is going to fit in a register, right?